### DISCUSSION TITLE
What does the `--lock` flag to `poetry check` do?

### DISCUSSION THREAD

**Discussion by allanlewis** [python-poetry/poetry#8495] [Q&A] [answered]
I see that `poetry lock --check` has been deprecated in favour of `poetry check --lock`. However, running `poetry check` locally gives the same output as `poetry check --lock`, so what does the `--lock` flag actually do? Is the intention to add more checks in the future?

Update for clarity: what I'm looking for is an explanation of the difference between `poetry check` and `poetry check --lock`.

**Comment by dimbleby** [comment]
try `poetry check --help`

**Reply by allanlewis** [reply]
> try `poetry check --help`

That's the first thing I did 🙂
Neither the usage docs nor the web docs explain the difference between `poetry check` and `poetry check --lock`.

**Reply by dimbleby** [reply]
read more carefully, it's right there.

**Reply by allanlewis** [reply]
```
$ poetry --version
Poetry (version 1.6.1)
$ poetry check --help

Description:
  Validates the content of the pyproject.toml file and its consistency with the poetry.lock file.

Usage:
  check [options]

Options:
      --lock                 Checks that poetry.lock exists for the current version of pyproject.toml.
...
```

So `poetry check` "Validates the content of the pyproject.toml file and its consistency with the poetry.lock file" and `poetry check --lock` "Checks that poetry.lock exists for the current version of pyproject.toml" - those sound like different commands.

I feel like I'm missing something here...

**Reply by finswimmer** [reply, marked as answer]
Without `--lock` `poetry check` succeeds if there is no  `poetry.lock` file at all. With `--lock` `poetry check` will fail if there is no lock file.

**Reply by allanlewis** [reply]
Thanks - given this wasn't clear to me from `--help` or the docs, perhaps an update is warranted?

**Reply by dimbleby** [reply]
please submit a merge request if you have an improvement
